src/content/blog/ddev-name-resolution-wildcards.md

```diff
@@ -0,0 +1,99 @@
+
+`ddev.site` and everything under it, like `you.ddev.site` and `something.else.entirely.ddev.site` are all (when you have working internet) records in the Domain Name System, maintained by the DDEV project. You don't have to think about them. You can `ping -c 1 something.ddev.site` and it will show that it is pinging `127.0.0.1`
+
+## What happens when you don't have internet?
+
+When you don't have internet capability, or DNS name resolution is broken, then DDEV's fallback during `ddev start` is to try to add the hostname to your `hosts` file. This is `/etc/hosts` on a Linux or macOS machine, or `C:\Windows\system32\drivers\etc\hosts` on a Windows machine. This is one of the very few times that DDEV will ever try to change the configuration of your local workstation. In general, the philosophy is never to do that. DDEV knows how to edit the hosts file properly and does it when you give it permission with your `sudo` password.
+
+If you *lose* internet after `ddev start` then your browser is going to struggle because it doesn't have a way to resolve the hostname. This can happen when you are working on a project and get on a plane, for example. `ddev restart` will force updating the hosts file, or `ddev hostname` can do it directly, or you can edit the hosts file.
+
+## Why does my project use `ddev.site` anyway?
+
+You don't have to use `ddev.site` as your "top-level domain name" or `TLD`. You can use anything you want... but we can't manage your DNS for you. For example, you can globally change your TLD to `example.com` with `ddev config global --project-tld=example.com` and DDEV will immediately start trying to use URLs like `https://yourproject.example.com`. If you prefer to use a domain name that you manage in your own local network DNS or in a DNS zone that you manage, you can do that too.
+
+## What happens with wildcard hostnames?
+
+DDEV supports wildcards in `additional_hostnames` in your `.ddev/config.yaml`. For example, this works great when you have internet and DNS working, because DDEV's `.ddev.site` will return `127.0.0.1` for any domain name you give it, including `a.b.c.d.e.f.g.h.ddev.site`.  So if you use `additional_hostnames: *.anything` then `a.anything.ddev.site` will work fine, as will `a.b.anything.ddev.site`. 
+
+However, wildcards don't work in `/etc/hosts` so DDEV has no way to provide a resolvable set of hostnames when you don't have internet work working DNS. As a result, you'll need to use explicit hostnames in this situation. For example, use `additional_hostnames: [a.anything,a.b.anything]`
+
+## Can I stop using the regular DNS setup with `ddev.site`?
+
+Of course. There are two easy ways:
+1. `use_dns_when_possible: false`: Turn off the use of DNS in your `.ddev/config.yaml`.
+2. Use a different `project_tld`. For example, if you use `project_tld: example.site` then DDEV will try to make your projects `<projectname>.example.site`. Since that is not resolvable, DDEV will resort to using hosts file manipulation.
```

Comment:
```suggestion
2. Use a different `project_tld`. If you are using for example `project_tld: example.site` then DDEV will try to use `<projectname>.example.site` for your projects. Since that is not resolvable, DDEV will resort to using hosts file manipulation.
```